### DISCUSSION TITLE
Feature request/help request: Early out from a load test if the first request fails

### DISCUSSION THREAD

**Discussion by scooper91** [artilleryio/artillery#2806] [Artillery]
We want to add an early-out to our load test - something as simple as it gets like if the first request fails, exit at that point and don't continue the test. We're very new to Artillery, so are trying to feel our way around the library.

We can't find an existing feature that does what we want. We tried to implement it ourselves with the before and after hooks, but either because of our naivety with the library or because of how artillery works with virtual users, we can't find any shared state across requests. We also considered whether we could have a phase which is just one request, and use something like the `ensure` plugin to exit at that point, but it doesn't look like it would solve our problem, partly because it doesn't seem that you can configure the `ensure` plugin per phase.

Any help or guidance would be appreciated, if it's already possible to early-out. Otherwise, it would be great if adding this as a feature could be considered.

Thanks,

Sophie

**Comment by hassy** [comment]
Hi @scooper91 👋 You should be able to do it in a `before` hook. You can run a [`function` step](https://www.artillery.io/docs/reference/engines/http#function-actions-and-beforescenario--afterscenario-hooks) which makes a request, and calls `process.exit()` if it fails (or on some other custom conditions)

It's not a very graceful solution, but it should work both for local runs and in distributed tests on Lambda/Fargate.
